### PR TITLE
[Krypton] Implement PVR API Timeshift functions and fix Kodi unable to play recently completed recordings

### PR DESCRIPTION
**Pull request by krustyreturns**
- Implement PVR API functions for timeshift/buffers
- Add workaround for Kodi not being able to play recently completed recordings due to cache issues

### CODE REVIEW

File: src/utilities.cpp [hunk 1]
```diff
@@ -81,6 +81,16 @@ bool StartsWith(CStdString const &fullString, CStdString const &starting)
     }
 }
 
+// return the directory from the input file path
+CStdString GetDirectoryPath(CStdString const &path)
```
Comment (ryangribble) [inline src/utilities.cpp]: This function doesnt seem to be being called by anything?

File: src/utilities.h [hunk 2]
```diff
@@ -29,5 +29,6 @@ bool Str2Bool(const CStdString str);
 
 bool EndsWith(CStdString const &fullString, CStdString const &ending);
 bool StartsWith(CStdString const &fullString, CStdString const &starting);
+CStdString GetDirectoryPath(CStdString const &path);
```
Comment (ryangribble) [inline src/utilities.h]: This function doesnt seem to be being called by anything?

Comment (krustyreturns) [inline src/utilities.h]: I wrote this function for an earlier fix of the advocate99 bug.  I ended up
doing the fix differently and didn't need this function anymore, but as a
utility function I thought it worth keeping, thinking we might want it in
the future.  I can remove it if you want.

 

  _____

**Comment by ryangribble** [comment]
Looks like nobody has any concerns so merging this now

**Pull request merged**